src/runner.py

```diff
@@ -99,34 +131,39 @@ def run(self):
         try:
             while True:
                 checkout_node = self._db.receive_node(sub_id)
-
-                self._logger.log_message(logging.INFO, "Tarball: {}".format(
-                    checkout_node['artifacts']['tarball']
-                ))
-
-                self._logger.log_message(logging.INFO, "Creating test node")
-                node = self._create_node(checkout_node)
-
-                tmp = tempfile.TemporaryDirectory(dir=self._output)
-                output_file = self._generate_job(node, tmp.name)
-
-                self._logger.log_message(logging.INFO, "Running test")
-                process = self._runtime.submit(output_file, get_process=True)
+                process, tmp = self._schedule_test(checkout_node)
                 self._job_tmp_dirs[process] = tmp
-
                 self._cleanup_paths()
         except KeyboardInterrupt as e:
             self._logger.log_message(logging.INFO, "Stopping.")
         finally:
             self._db.unsubscribe(sub_id)
             self._cleanup_paths()
 
+    def run(self):
+        if self._node_id:
+            checkout_node = self._db.get_node(node_id)
+            self._run_single_job(self._checkout_node)
+        elif self._git_commit:
+            node = self._get_node_from_commit(self._git_commit)
```

Comment:
```suggestion
            checkout_node = self._get_node_from_commit(self._git_commit)
            if not checkout_node:
                self._logger.log_message(logging.ERROR,
                                         f"Node not found with commit hash:\
{self._git_commit}")
                return
            self._run_single_job(checkout_node)
```
Also, `node` can be renamed to `checkout_node`.

Reply:
Yes. But I don't like the idea of adding all the error handling code here again.

Reply:
I think I'll add exception handling in #59 instead which has a separate method for manual runs.

Reply:
Yes, we need to catch this too:
```
'tarball_url': node['artifacts']['tarball'],
TypeError: 'NoneType' object is not subscriptable
```

Reply:
Actually there's already some logic in place in this commit from #59 to return an error if the checkout node wasn't found using the command line arguments:
https://github.com/kernelci/kernelci-pipeline/pull/59/commits/49210ba5590cbb64012ec703a93064e50862eac5
```python
   def run(self):
        if self._node_id:
            checkout_node = self._db.get_node(self._node_id)
            self._run_single_job(checkout_node)
        elif self._git_commit:
            checkout_node = self._get_node_from_commit(self._git_commit)
            self._run_single_job(checkout_node)
        else:
            node = None

        if node is None:
            return False

        plan_config = self._plan_configs[args.plan]
        device_config = self._device_configs[args.target]
        return self._run_single_job(node, plan_config, device_config)
```
So I could add something there to print an error message when there's no node, and also another commit to handle exceptions for any unexpected issues.  If that's OK then we can merged this current PR #50 and I'll add the error handling improvements in #59.

Reply:
Sounds good to me.